server/rooms.ts

```diff
@@ -51,6 +51,29 @@ interface BattleRoomTable {
 	minElo?: 'tour' | number;
 }
 
+interface LastTournamentData {
+	type: string;
+	rootNode: LastTournamentDataNode;
+	playersLength: number;
+}
+
+export type LastTournamentDataNode = LastTournamentDataBattle | LastTournamentDataTeam;
+
+interface LastTournamentDataBattle {
+	children: LastTournamentDataNode[];
+	state: string;
+	team: string;
```

Comment:
`team` doesn't actually mean "team", does it? This should maybe be documented.

Also this should probably be typed in Tours itself, if we're going to type it.

Reply:
Also I'd just call it `TournamentData`; there's nothing "last" about the type.

Reply:
It's not the data of the current Tournament though, it's the data of the Last one that was completed. I can change it to TournamentData though. 
Team is the term that was already used in the Object, I'm not changing anything here, just reflecting the data I'm given.

Reply:
No, it's the data of any tournament. You're mixing up the data type with the data itself.

Reply:
I know you can't rename `team`, but you can document that it's poorly named.

Reply:
Oh I see what you're saying. There is only one use case for this data type though. But I can make the change. And yes, I'll comment

Reply:
No, there's another use: as the return type of `getBracketData`. That's also why I think this type should be defined in `tournaments/index`.

Reply:
It's not quite the return type of getBracketData, I think I'm also including a PlayerCount in there for ease of calculation (I didn't want to have to traverse the tree to count the player numbers). I manually insert this playercount in onTournamentEnd()

Reply:
Maybe it should be. But the other types _are_ used in `getBracketData`, anyway.

Reply:
I would probably just drop `playerCount` from the type, and add it back with `& {playerCount: number}` in the variable definition.

Reply:
Not sure I know what you mean exactly? sorry

Reply:
Um, I'd just do this for you but I'm really backlogged right now so it might take a dayish.

Reply:
see https://www.typescriptlang.org/docs/handbook/advanced-types.html#intersection-types

Remove the `playerCount` field from the type and instead declare the variable to be `LastTournamentDataBattle & {playerCount: number}`

Reply:
thanks!